api/v1alpha1/etcdcluster_types.go

```diff
@@ -50,10 +53,55 @@ type ProviderConfig struct {
 	CertManagerCfg *ProviderCertManagerConfig `json:"certManagerCfg,omitempty"`
 }
 
+type AltNames struct {
+	// DNSNames accepts a string array, if nil programmatically populates the
+	// CommonName as DNSNames
+	// +optional
+	DNSNames []string `json:"dnsNames,omitempty"`
+
+	// IPs accepts a net.IP array of IP Addresses as alternative names
+	// +optional
+	IPs []net.IP `json:"ipAddresses,omitempty"`
+}
+
+type CommonConfig struct {
+	// CommonName accepts a string for a human-readable name of the certificate
+	// +optional
+	CommonName string `json:"commonName,omitempty"`
+
+	// Organization accepts a string array to be used on the Certificate.
+	// +optional
+	Organization []string `json:"organizations,omitempty"`
+
+	// AltNames contains the domain names and IP addresses that will be added
+	// to the x509 certificate SubAltNames fields. The values will be passed
+	// directly to the x509.Certificate object.
+	AltNames AltNames `json:"altNames,omitempty"`
+
+	// ValidityDuration accepts time.Duration, if nil defaults to 90 days
+	// +optional
+	ValidityDuration time.Duration `json:"validityDuration,omitempty"`
+
+	// CABundleSecret accepts a secret name with CABundle present
+	// if nil, will consider IssuerKind, IssuerName for certificate creation
```

Comment:
In golang, a string can never be a `nil`
```suggestion
	// if empty, will consider IssuerKind, IssuerName for certificate creation
```

Reply:
It seems that comment isn't valid based on https://github.com/etcd-io/etcd-operator/pull/103#discussion_r2070201524

Reply:
Yes, re-writing the comment based on the discussion.